service/controller/clusterapi/v29/resource/machinedeploymentazs/resource.go

```diff
@@ -0,0 +1,50 @@
+// Package machinedeploymentazs implements a resource to gather all private
+// subnets for the configured availability zones of a node pool. Like the
+// clusterazs resource, we need logic to take the node pool subnet allocated by
+// the ipam resource and split it according to the configured availability
+// zones. We then have 1, 2 or 4 private subnet CIDRs we put into the controller
+// context for further use in the tcnp resource. Note that the availability
+// zones of a node pool cannot be updated upon creation due to the network
+// splitting. In order to change availability zones one must delete and create
+// node pools accordingly.
```

Comment:
I found the description in the PR useful and put an extended version here in the package godoc.

Reply:
Very nice, thx!